docs/mp/reactivemessaging/mock.adoc

```diff
@@ -0,0 +1,140 @@
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Mock Connector
+:toc:
+:toc-placement: preamble
+:description: Reactive Messaging Mock connector for testing
+:keywords: helidon, mp, messaging, test, mock
+:feature-name: Mock Connector
+:microprofile-bundle: false
+:rootdir: {docdir}/../..
+
+== ToC
+
+- <<Overview, Overview>>
+- <<Emitting data, Emitting data>>
+- <<Asserting data, Asserting data>>
+- <<Configuration, Configuration>>
+- <<Helidon test with Mock Connector, Helidon Test>>
+
+== Overview
+Mock connector is a simple application scoped bean that can be used for emitting to a channel
+or asserting received data in a test environment. All data received are kept in memory only.
+
+include::{rootdir}/includes/mp.adoc[]
+include::{rootdir}/includes/dependencies.adoc[]
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.messaging.mock</groupId>
+    <artifactId>helidon-messaging-mock</artifactId>
+</dependency>
+----
+
+WARNING: Mock connector should be used in the test environment only!
+
+For injecting Mock Connector use `@TestConnector` qualifier:
+[source,java]
+----
+@Inject
+@TestConnector
+MockConnector mockConnector;
+----
+
+=== Emitting data
+
+.Emitting String values `a`, `b`, `c`
+[source,java]
+----
+mockConnector
+     .incoming("my-incoming-channel", String.class) // <1>
+     .emit("a", "b", "c");
+----
+<1> Get incoming channel of given name and payload type
+
+=== Asserting data
+
+.Awaiting and asserting payloads with custom mapper
+[source,java]
+----
+mockConnector
+    .outgoing("my-outgoing-channel", String.class)           // <1>
+    .awaitData(TIMEOUT, Message::getPayload, "a", "b", "c"); // <2>
+----
+<1> Get outgoing channel of given name and payload type
+<2> Request number of expected items and block the thread until items arrive then assert the payloads
+
+=== Configuration
+|===
+|Key            |Default value   | Description
+|mock-data      |                | Initial data emitted to the channel immediately after subscription
+|mock-data-type |java.lang.String| Type of the emitted initial data to be emitted
+|===
+
+== Helidon test with Mock Connector
+Mock connector works great with built-in Helidon test support for
+link:/{rootdir}/testing/testing.adoc[JUnit 5] or link:/{rootdir}/testing/testing-ng.adoc[Test NG].
+
+As Helidon test support makes a bean out of your test, you can inject MockConnector right in to it.
```

Comment:
Mock Connector directly into it or is it MockConnector when referring to the bean?